demo-ehr-app/src/pages/MedicalImaging.tsx

```diff
@@ -0,0 +1,460 @@
+import { Link } from "react-router-dom";
+import { LAB_TEST, ORDER_SIGN_CDS_REQUEST2 } from "../constants/data";
+import { ExpandedContext } from "../utils/expanded_context";
+import { Card } from "@chakra-ui/react";
+import axios from "axios";
+import { updateCdsResponse } from "../redux/cdsResponseSlice";
+import MedicalImagingSign from "./MedicalImagingSign";
+
+interface Coding {
+  id?: string;
+  extension?: object;
+  system?: string;
+  version?: string;
+  code?: string;
+  display?: string;
+  userSelected?: boolean;
+}
+
+enum ActionType {
+  CREATE = "create",
+  UPDATE = "update",
+  DELETE = "delete",
+}
+
+interface Action {
+  type: ActionType;
+  description: string;
+  resource?: object;
+  resourceId?: string;
+}
+
+interface Source {
+  label: string;
+  url?: string;
+  icon?: string;
+  topic?: Coding;
+}
+
+interface Suggestion {
+  label: string;
+  uuid?: string;
+  isRecommended?: boolean;
+  actions?: Action[];
+}
+
+enum LinkType {
+  ABSOLUTE = "absolute",
+  SMART = "smart",
+}
+
+export interface CdsLink {
+  label: string;
+  url: string;
+  type: LinkType;
+  appContext?: string;
+}
+
+export interface CdsCard {
+  uuid?: string;
+  summary: string;
+  detail?: string;
+  indicator: string;
+  source: Source;
+  suggestions?: Suggestion[];
+  selectionBehavior?: string;
+  overrideReason?: Coding[];
+  links?: CdsLink[];
+}
+
+export interface CdsResponse {
+  cards: CdsCard[];
+  systemActions?: Action[];
+}
+
+function MedicalImaging() {
+  const form_selector_width = SCREEN_WIDTH * 0.3;
+  const { expanded } = useContext(ExpandedContext);
+  const dispatch = useDispatch();
+  const selectedPatientId = useSelector(
+    (state: any) => state.patient.selectedPatientId
+  );
+
+  const [selectedTestType, setSelectedTestType] = useState("");
+  const [selectedArea, setSelectedArea] = useState("");
+  const [selectedDate, setSelectedDate] = useState<Dayjs | null>(null);
+  const [selectedSlots, setSelectedSlots] = useState("");
+  const [selectedTreatingDisease, setSelectedTreatingDisease] = useState("");
+
+  const [isTestTypeChanged, setIsTestTypeChanged] = useState(false);
+  const [isSelectedAreaChanged, setIsSelectedAreaChanged] = useState(false);
+  const [isSelectedDateChanged, setIsSelectedDateChanged] = useState(false);
+  const [isSelectedSlotsChanged, setIsSelectedSlotsChanged] = useState(false);
+  const [
+    isSelectedTreatingDiseaseChanged,
+    setIsSelectedTreatingDiseaseChanged,
+  ] = useState(false);
+
+  const [enableSubmitButton, setEnableSubmitButton] = useState(false);
+  const [enableNextStep, setEnableNextStep] = useState(false);
+  const [isCrdCheckFinished, setCrdCheckFinished] = useState(false);
+  const [cdsCards, setCdsCards] = useState<CdsCard[]>([]);
+
+  const validateError = (
+    <>
+      <Alert severity="error" sx={{ marginLeft: 1 }}>
+        This is an mandotary field.
+      </Alert>
+    </>
+  );
+
+  const [patientId] = useState("PA2347");
```

Comment:
Don't we need to keep the IDs in `data.ts` file?

Reply:
In the data.ts file, we have templates or placeholders to parse the dynamic data like patientId and practinionerId in the request payload.